### ISSUE TITLE
Spot nodes launch fail at China region

### ISSUE THREAD

**Issue by xchan** [aws/karpenter#2896]
### Version

__Karpenter Version:__ v0.19.1

__Kubernetes Version:__ v1.24.0

### Expected Behavior

Karpenter should work when scale-up using spot instance.

I believe currently China region only support 4 types allocation strategies for spot Instances, and the docs differ between Chinese and English version.
https://docs.amazonaws.cn/AWSEC2/latest/UserGuide/spot-fleet-allocation-strategy.html <-- indicating 4 types
https://docs.amazonaws.cn/en_us/AWSEC2/latest/UserGuide/spot-fleet-allocation-strategy.html <-- indicating 5 types

### Actual Behavior

launch template will report error

### Steps to Reproduce the Problem

Launch EKS in aws-cn partition, install karpenter with provitioner settings include
      - key: karpenter.sh/capacity-type
        operator: In
        values: ["spot"]

### Resource Specs and Logs

2022-11-21T10:37:11.630Z	INFO	controller.provisioning	Found 5 provisionable pod(s)	{"commit": "ea5dc14"}
2022-11-21T10:37:11.630Z	INFO	controller.provisioning	Computed 1 new node(s) will fit 5 pod(s)	{"commit": "ea5dc14"}
2022-11-21T10:37:11.630Z	INFO	controller.provisioning	Launching node with 5 pods requesting {"cpu":"447m","memory":"1230Mi","pods":"9"} from types t3.large, t3.xlarge	{"commit": "ea5dc14", "provisioner": "default"}
2022-11-21T10:37:11.786Z	ERROR	controller.provisioning	launching node, creating cloud provider instance, creating fleet InvalidParameterValue: AllocationStrategy must be "lowest-price", "capacity-optimized", "capacity-optimized-prioritized", or "diversified"
	status code: 400, request id: 3db22e66-ea16-4d14-a390-2ff5a9858ab8 (3db22e66-ea16-4d14-a390-2ff5a9858ab8)	{"commit": "ea5dc14"}

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by xchan** [comment]
Can we have a dynamic section in provitioner?

**Comment by hitsub2** [comment]
According  to the [EC2 Spot annoucement](https://docs.amazonaws.cn/en_us/aws/latest/userguide/ec2-auto-scaling.html#feature-diff), 
```docs
"Specifying price-capacity-optimized for the SpotAllocationStrategy property of a mixed instances group 
is currently not supported in the Beijing and Ningxia Regions.
```
Just wondering how karpenter takes care of this diff for now and feature. It is really confusing us when adopting karpenter.

**Comment by bwagner5** [comment]
Apologies for this! We did not realize the new `priceCapacityOptimized` was not launched in the China regions yet. Currently, we'd recommend downgrading to v0.18.1 until we get a patch out to remove the new allocation strategy for China regions until it is released there by EC2.

**Comment by bwagner5** [comment]
We will likely not support #1240 and instead take a strong opinion on which allocation strategy should be used for capacity-types. Right now, `price-capacity-optimized` is the recommended spot allocation strategy for the vast majority of use-cases and OD is `lowest-price`.  Karpenter was approximating what `price-capacity-optimized` does with our implementation of `capacity-optimized-prioritized` by specifying the priorities as the ranked cost of instance types by consuming the spot pricing data.

**Comment by bwagner5** [comment]
Fixed in v0.19.2

**Comment by xchan** [comment]
Verfied fix with scale up nodes at AWS-CN-Ningxia Region
